pkg/k8s/version/version.go

```diff
@@ -271,39 +212,36 @@ func updateK8sServerVersion(client kubernetes.Interface) error {
 func Update(client kubernetes.Interface, apiDiscoveryEnabled bool) error {
 	err := updateK8sServerVersion(client)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to update Kubernetes server version: %s", err)
 	}
 
-	if apiDiscoveryEnabled {
-		// Discovery of API groups requires the API services of the
-		// apiserver to be healthy. Such API services can depend on the
-		// readiness of regular pods which require Cilium to function
-		// correctly. By treating failure to discover API groups as
-		// fatal, a critical loop can be entered in which Cilium cannot
-		// start because the API groups can't be discovered and th API
-		// groups will only become discoverable once Cilium is up.
-		_, apiResourceLists, err := client.Discovery().ServerGroupsAndResources()
-		if err != nil {
-			// It doesn't make sense to retry the retrieval of this
-			// information at a later point because the capabilities are
-			// primiarly used while the agent is starting up. Instead, fall
-			// back to probing API endpoints directly.
-			log.WithError(err).Warning("Unable to discover API groups and resources")
-			if err := endpointSlicesFallbackDiscovery(client); err != nil {
-				return err
-			}
-
-			return leasesFallbackDiscovery(client, apiDiscoveryEnabled)
+	if !apiDiscoveryEnabled {
+		if err := endpointSlicesFallbackDiscovery(client); err != nil {
+			return fmt.Errorf("failed to fallback-discover endpoint slices: %s", err)
 		}
+		return nil
+	}
 
-		updateServerGroupsAndResources(apiResourceLists)
-	} else {
+	// Discovery of API groups requires the API services of the
+	// apiserver to be healthy. Such API services can depend on the
+	// readiness of regular pods which require Cilium to function
+	// correctly. By treating failure to discover API groups as
+	// fatal, a critical loop can be entered in which Cilium cannot
+	// start because the API groups can't be discovered and th API
```

Comment:
typo: "and th API"

Reply:
Just realized this is existing code, but we might as well clean these up if we're touching these 😄